File: server/dna.core/dna.core.service/Models/Abstract/WriteHistoryBaseModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace dna.core.service.Models.Abstract
{
    public abstract class WriteHistoryBaseModel
    {
        [Required]
        public int CreatedById { get; set; }

        [Required]
        public int UpdatedById { get; set; }
        public virtual UserModel CreatedByUser { get; set; }


        public virtual UserModel UpdatedByUser { get; set; }

        public DateTime? CreatedDate { get; set; }
        public DateTime? UpdatedDate { get; set; }
    }
}

File: server/medicore.admin/src/app/model/writeHistoryBaseModel.ts
import { Type }                 from 'class-transformer';
import { UserModel }         from './userModel';
export class WriteHistoryBaseModel{
        createdById:number;
        updatedById:number;
        @Type(() => UserModel)
        createdByUser:UserModel;
        @Type(() => UserModel)
        updatedByUser:UserModel;        
        createdDate:Date;
        updatedDate:Date;

        constructor(){
            this.createdById = 0;
            this.updatedById = 0;
            this.createdDate = null;
            this.updatedDate =  null;
            this.createdByUser = null;
            this.updatedByUser = null;
        }
       
}